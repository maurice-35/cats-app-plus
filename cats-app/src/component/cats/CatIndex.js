import React, { useState, useEffect } from "react";
import axios from "axios";
import { Row, CardGroup, Card } from "react-bootstrap";
import CatCard from "./CatCard";
import ScrollToTop from "react-scroll-to-top";

const App = () => {
  const [cat, setCat] = useState([]);
  const [hasError, setHasError] = useState(false);

  useEffect(() => {
    const getData = async () => {
      try {
        const { data } = await axios.get("https://api.thecatapi.com/v1/breeds");
        console.log("DATA", data);
        setCat(data);
      } catch (err) {
        setHasError(true);
      }
    };
    getData();
  }, []);

  return (
    <div>
      <Row className="grid-container">
        <CardGroup>
          <Card>
            {cat.length > 0 ? (
              <CardGroup>
                <Card id="cat-map">
                  {cat.map((breed) => (
                    <CatCard key={breed.id} {...breed} />
                  ))}
                </Card>
              </CardGroup>
            ) : (
              <h2 className="index">
                {hasError ? "Please wait" : "...loading"}
              </h2>
            )}
          </Card>
        </CardGroup>
      </Row>
      <ScrollToTop smooth />
    </div>
  );
};

export default App;
