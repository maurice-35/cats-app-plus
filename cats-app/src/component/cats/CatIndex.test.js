import { screen, render, fireEvent } from '@testing-library/react'
import { unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import CatIndex from './CatIndex'

const mockedSetCatIndex = jest.fn()

describe("CatCard", () => {
	it("should display loading page", () => {
		render(
			<CatIndex
				cat={[]}
				setCat={mockedSetCatIndex}
			/>
		)
		const displayDescription = screen.getByText(/...loading/i)
		expect(displayDescription).toMatchSnapshot()
	})
})


let container = null
beforeEach(() => {
	container = document.createElement("div")
	document.body.appendChild(container)
})

afterEach(() => {
	unmountComponentAtNode(container)
	container.remove()
	container = null
})
it('renders breed data', async () => {
		const fakeBreed = {
			image: "",
			origin: "",
			name: "",
			description: "Lovely cat",
			energy_level: 4,
			child_friendly: 3,
			dog_friendly: 2,
			intelligence: 1,
			social_needs: 3,
			health_issues: 2,
			adaptability: 4,
			vocalisation: 3,
			affection_level: 2,
			temperament: 'Active',
			grooming: 3,
			life_span: 6 - 7,
			weight: "",
		}
		jest.spyOn(global, "fetch").mockImplementation(() => 
		Promise.resolve({
			json: () => Promise.resolve(fakeBreed)
		})
		)
		
		// eslint-disable-next-line testing-library/no-unnecessary-act
		await act(async () => {
			fireEvent.click(container)
			render(<CatIndex id="123" />, container)
		})

		expect(container.textContent).toContain(fakeBreed.name)
		expect(container.textContent).toContain(fakeBreed.origin)
		expect(container.textContent).toContain(fakeBreed.image)
		expect(container.textContent).toContain(fakeBreed.weight)
		global.fetch.mockRestore()
})
