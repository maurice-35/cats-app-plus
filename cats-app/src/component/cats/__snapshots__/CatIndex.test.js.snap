// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatCard should display loading page 1`] = `
<h2
  class="index"
>
  ...loading
</h2>
`;
